connector/connect/client/src/main/scala/org/apache/spark/sql/connect/client/SparkSession.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connect.client
+
+import scala.language.existentials
+
+import io.grpc.{ManagedChannel, ManagedChannelBuilder}
+import org.apache.arrow.memory.RootAllocator
+
+import org.apache.spark.SPARK_VERSION
+import org.apache.spark.connect.proto
+
+
+class SparkSession(
```

Comment:
From our work in the Python client it turned out that it's actually better to keep the GRPC client out of the `SparkSession` and have a separate `SparkConnectClient` class for that. That gives you as well better testability.

Reply:
great +1 on this.

Reply:
Why? What do you want to test?

Reply:
There is a difference in testing the GRPC client and the SparkSession, for some parts we inject a mock client into the session because we don't want to or cannot setup the client. 

That's just a very reasonable separation of concerns. The SparkSession deals with the SparkSide, while the client deals with the proto side.